### ISSUE TITLE
Order that coloured lights are mixed is arbitrarily decided on scene load

### ISSUE THREAD

**Issue by Avantir-Chaosfire** [godotengine/godot#43750]
**Godot version:**
3.2.3

**OS/device including version:**
Mac OS X 10.11.6, Windows 10

**Issue description:**
When you put two coloured lights into mix mode and have them overlap, the resulting light pattern is arbitrary. I don't know how this is handled internally, but from my tests it seems like the algorithm is basically just overlapping the translucent light images to get the light pattern. With light of all the same colour, this is fine. But when the lights are different colours the order matters. Example:

Red on top:

![Red on top](https://user-images.githubusercontent.com/11003232/99882239-dc5b5900-2bd3-11eb-821c-f451e93f1f02.png)

Green on top:

![Green on top](https://user-images.githubusercontent.com/11003232/99882248-e67d5780-2bd3-11eb-8310-662cb5214720.png)

Minimal reproduction project is linked below. You can hold the enter key to continually reload the scene, and you'll see it flicker back and forth between each colour being on top. It might take a few seconds to change, and stopping on a particular pattern can be tricky.

One possible fix is to allow the user to determine the order the lights are mixed - e.g. they're ordered according to the order of their nodes in the scene tree. Another possible fix is to change the mix lighting algorithm such that order DOESN'T matter - this is, I believe, the behaviour people will expect when seeking out mixed coloured lighting. It doesn't make sense for one light to override another.

**Minimal reproduction project:**
https://github.com/Avantir-Chaosfire/MixColouredLightingBug

**Comment by Calinou** [comment]
Does it improve if you change the **Process Priority** of one of the nodes?

**Comment by Avantir-Chaosfire** [comment]
@Calinou Process Priority does not appear to make any difference.

**Comment by filipinyo** [comment]
What's the status with this? I believe I'm encountering the same issue on 4.2.

**Comment by Calinou** [comment]
> I believe I'm encountering the same issue on 4.2.

Could you upload a minimal reproduction project? I can't reproduce this issue with any rendering method using https://github.com/Avantir-Chaosfire/MixColouredLightingBug converted to Godot 4.

**Comment by AThousandShips** [comment]
Closing due to lack of response, if you provide the requested information this can be reopened

**Comment by lewismmason** [comment]
I am currently having the exact same issue, except my case specifically acts on lights that work like masks. I have tried changing both process and physics process order, as well as the visibility order and it is completely arbitrary for the order. I also found that this happends when using "add" lighting. For this example I show mixed lighting.

Here are two runs:

Expected behaviour:
![Screenshot 2024-09-29 113854](https://github.com/user-attachments/assets/ee15d72e-95d1-4f55-900e-4124768bf922)

Random behaviour:
![image](https://github.com/user-attachments/assets/e18f6824-59a3-445b-b53a-ebb34075b6e4)
![image](https://github.com/user-attachments/assets/28e390db-9b25-4901-ae38-b534283f686a)

I want to have even more lights in the game. But ALL of them do this where they are ordered wrong. The issue is a secondary ring that I need for low visibility light. The other brighter light sources will randomly go behind or in front of this lower visibility light, even though it is mixed. This occurs regardless of process priority, physics process orders or of visibility orders.

The lights cut out from a canvas modulute over the entire game that makes stuff black.
